src/apps/chatter/hooks/use-room.js

```diff
@@ -0,0 +1,92 @@
+import url from '/util/url'
+import { useRef, useEffect, useState } from 'react'
+import { getState } from '/store'
+
+export function useRoom (roomId) {
+  const [messages, setMessages] = useState([])
+  const socketRef = useRef()
+
+  useEffect(() => {
+    connect()
+  }, [])
```

Comment:
The only feedback i have is, should we cleanup/remove the websocket connection on unmount? Something like:

```javascript
  useEffect(() => {
    connect()
    return () => socketRef.current && socketRef.current.disconnect() // or whatever the websocket API is to close/disconnect?
  }, [])
```